tal/render/render.py

```diff
@@ -1,443 +1,707 @@
-            scene_config['sensor_x'],
-            scene_config['sensor_y'],
-            scene_config['sensor_z'],
-        ], dtype=np.float32)
-        capture_data.sensor_grid_xyz = sensor_grid_xyz
-        capture_data.sensor_grid_normals = sensor_grid_normals
-        capture_data.sensor_grid_format = GridFormat.X_Y_3
-        capture_data.laser_xyz = np.array([
-            scene_config['laser_x'],
-            scene_config['laser_y'],
-            scene_config['laser_z'],
-        ], dtype=np.float32)
-        capture_data.laser_grid_xyz = laser_grid_xyz
-        capture_data.laser_grid_normals = laser_grid_normals
-        capture_data.laser_grid_format = GridFormat.X_Y_3
-        # NOTE(diego): we do not store volume information for now
-        # capture_data.volume_format = VolumeFormat.X_Y_Z_3
-        capture_data.delta_t = scene_config['bin_width_opl']
-        capture_data.t_start = scene_config['start_opl']
-        capture_data.t_accounts_first_and_last_bounces = \
-            scene_config['account_first_and_last_bounces']
-        capture_data.scene_info = {
-            'tal_version': tal.__version__,
-            'config': scene_config,
-            'args': vars(args),
-        }
-        if scan_type == 'single':
-            hdr_path, _ = mitsuba_backend.partial_laser_path(
-                partial_results_dir,
+        if simulation_params is None:
+            return  # Skip
+
+        run_mitsuba_f = simulation_params.make_mitsuba_call(self.mitsuba_backend)
+
+        if platform.system() == "Windows":
+            # Windows does not support multiprocessing
+            run_mitsuba_f()
+        else:
+            process = multiprocessing.Process(target=run_mitsuba_f)
+            try:
+                process.start()
+                process.join()
+            except KeyboardInterrupt:
+                process.terminate()
+                raise KeyboardInterrupt
+        self._teardown_single_simulation(simulation_params, i)
+
+        if self.scan_type == "exhaustive" and i == 0:
+            size_bytes = os.path.getsize(simulation_params.hdr_path)
+            final_size_gb = size_bytes * len(self.laser_lookats) / 2**30
+            pbar.set_description(
+                f"Rendering {self.experiment_name} ({self.scan_type}, estimated size: {final_size_gb:.2f} GB)..."
+            )
+
+    def _setup_single_simulation(
+        self, experiment_name, i, laser_lookat: Point2D
+    ) -> Optional[SimulationParams]:
+        try:
+            hdr_path, is_dir = self.mitsuba_backend.partial_laser_path(
+                self._get_directory("partial_results_dir"),
                 experiment_name,
-                *laser_lookats[0])
-            capture_data.H = mitsuba_backend.read_transient_image(hdr_path)
+                laser_lookat.x,
+                laser_lookat.y,
+            )
+            # ToDo: why do we have `self.args.quiet` here?
+            if os.path.exists(hdr_path) and not self.args.quiet:
```

Comment:
It's probably because of some legacy interaction between args.quiet and args.dry_run, which when used with mitsuba2 prints the `mitsuba` command used. This was useful for debugging (you want to print the command even if the image exists). But mitsuba3 does not work using commands, and probably this code was wrong before lol, let's just remove it
```suggestion
            if os.path.exists(hdr_path):
```

Reply:
Got it, thanks :)